src/group_theory/burnside.lean

```diff
@@ -0,0 +1,153 @@
+  rw [mul_assoc,mul_inv_cancel_right,mul_inv_cancel_left]
+end
+
+def equiv.stabilizer (g : G) (x : α) : stabilizer G x ≃ stabilizer G (g • x) :=
+begin
+  use assume g', ⟨g * g'.1 * g⁻¹, stabilizer_orbit_iff.1 g'.2⟩,         -- to_fun
+  use assume g', ⟨g⁻¹ * g'.1 * g, stabilizer_orbit_inv_iff.1 g'.2⟩,     -- inv_fun
+  all_goals { assume g',
+              apply subtype.eq',
+              simp,
+              rw mul_assoc },
+    { rw inv_mul_cancel_right,
+      exact inv_mul_cancel_left _ _ },
+    { rw mul_inv_cancel_right,
+      exact mul_inv_cancel_left _ _ }
+end
+
+variables G α
+def orbits : Type v := quotient $ orbit_rel G α
+
+local attribute [instance] mul_action.orbit_rel
+local attribute [reducible] lattice.fixed_points
+
+/-- A trivial equivalence used in the proof of Burnside Lemma -/
+def burnside_aux₁ :
+  (Σ g : G, lattice.fixed_points ((•) g : α → α)) ≃
+    (Σ (o : orbits G α) (x : {x // ⟦x⟧ = o}), stabilizer G x.val) :=
+{ to_fun  := λ y, ⟨⟦y.snd.val⟧, ⟨⟨y.snd.val, rfl⟩, ⟨y.fst, y.snd.property⟩⟩⟩,
+  inv_fun := λ y, ⟨y.snd.snd.val, ⟨y.snd.fst.val, y.snd.snd.property⟩⟩,
+  left_inv := λ ⟨g, ⟨x, h⟩⟩, rfl,
+  right_inv := λ ⟨o, ⟨⟨x, hx⟩, ⟨g, h⟩⟩⟩, by subst o
+}
+
+noncomputable theory
+open classical
+
+variables {G α}
+
+/-- Choose an element of the group that sends `o.out` to a specific point of the orbit. -/
+def orbit_choice {o : orbits G α} (x : {x // ⟦x⟧ = o}) : G :=
+  some (quotient.exact $ x.2.trans o.out_eq.symm)
+
+def orbit_choice_spec {o : orbits G α} (x : {x // ⟦x⟧ = o}) :
+  orbit_choice x • o.out = x.val :=
+some_spec (quotient.exact $ x.2.trans o.out_eq.symm)
+
+/-- Disjoint union of stabilizers of all points of a given orbit
+    has the same cardinality as the group. -/
+def sigma_stabilizer_equiv_group (o : orbits G α) :
+  (Σ x : {x // ⟦x⟧ = o}, stabilizer G x.val) ≃ G :=
+begin
+  have h₁ : (Σ x : {x // ⟦x⟧ = o}, stabilizer G x.val) ≃
+    ({x // ⟦x⟧ = o} × (stabilizer G o.out)),
+    by { apply equiv.sigma_equiv_prod_of_equiv,
+         assume x,
+         rw [← (orbit_choice_spec x)],
+         symmetry,
+         apply equiv.stabilizer },
+
+  have h₂ : ({x // ⟦x⟧ = o} × (stabilizer G o.out)) ≃ ((orbit G o.out) × (stabilizer G o.out)),
+    by { apply equiv.cast,
+         congr,
+         funext,
+         apply propext,
+         rw [← quotient.out_eq o,quotient.eq,quotient.out_eq],
+         refl },
+
+  have h₃ : ((orbit G o.out) × (stabilizer G o.out)) ≃ G,
+    by { generalize: o.out = y,
+         symmetry,
+         apply (@mul_action.is_subgroup G _ _ _ y).group_equiv_quotient_times_subgroup.trans,
+         apply equiv.prod_congr,
+           { exact (orbit_equiv_quotient_stabilizer _ _).symm },
+           { refl } },
+
+  calc _ ≃ ({x // ⟦x⟧ = o} × (stabilizer G o.out))  : h₁
+    ...  ≃ ((orbit G o.out) × (stabilizer G o.out)) : h₂
+    ...  ≃ G                                        : h₃
+end
+
+/-- Equivalence behind Burnside Lemma.
```

Comment:
These docstrings show up in mouse-over tooltips in VScode. It is very good that you write them! However, I suggest that you compactify this a little bit. Maybe replace the `.` with a `:` and continue your explanation on the same line, instead of leaving an empty line.

Reply:
I'll do it. BTW, (a) are they integrated into Emacs `lean-mode`? (b) What do you think about adopting Doxygen style (brief description in the first line + detailed discussion after a blank like)? Tooltips can show the first line, and expand it on click/hover.

Reply:
@digama0 You're the boss concerning the style guide... what's the rule :wink: ?

Reply:
I'm okay with longer comments; at least in vscode it's not a big deal to have a decent amount of text here, even up to 10-20 lines. The first line is not treated specially by emacs or vscode, but I think that summarizing in the first sentence is a good style regardless, with or without a paragraph break.

Reply:
I hope, comments look better now.